applications/spark/README.md

```diff
@@ -2,51 +2,170 @@
-- Spark 3.3.1 and Python 3.9 is at `/datasets/images/apache_spark/spark_py39.sif`.
-This image includes the packages `ipython`, `jupyter`, `numpy`, `pandas`, and `pyarrow`.
-- Spark 3.3.1 and R 4.0.4 is at `/datasets/images/apache_spark/spark_r.sif`.
-This image includes the packages `tidyverse`, `sparklyr`, `data.table`, `here`, `janitor`, and
-`skimr`.
+**Note**: If you are running on Eagle, replace `apptainer` with `singularity`. It was rebranded
+and Eagle has the old software.
 
-## Setup
-
-1. Clone the repository:
-```
-$ git clone https://github.com/NREL/HPC.git
+Kestrel:
 ```
-2. Change to a directory in `/scratch/$USER`.
-3. Add the `spark_scripts` directory to your path.
+$ module load apptainer
+$ apptainer --help
 ```
-$ export PATH=$PATH:<your-repo-path>/HPC/applications/spark/spark_scripts
+Eagle:
 ```
-4. Copy the `config` file and `conf` directory with the command below. Specify an alternate
-destination directory with `-d <directory>`. This can be done on the login node or compute node.
-```
-$ create_config.sh -c <path-to-spark-container>
+$ module load singularity-container
+$ singularity --help
 ```
 
-5. Edit the `config` file if necessary. Note that the rest of this page relies on the setting
+Existing containers on Kestrel and Eagle:
+
+- Spark 3.5.0 and Python 3.11, 3.12 (default = 3.12):
+  - This image includes the packages `ipython`, `jupyter`, `numpy`, `pandas`, and `pyarrow`.
+  - Kestrel: `/kfs2/pdatasets/images/apache_spark/spark350_py311.sif`
+  - Eagle: `/datasets/images/apache_spark/spark350_py311.sif`
+
+- Spark 3.3.1 and R 4.0.4:
+  - This image includes the packages `tidyverse`, `sparklyr`, `data.table`, `here`, `janitor`, and
+    `skimr`.
+  - Kestrel: `/kfs2/pdatasets/images/apache_spark/spark_r.sif`
+  - Eagle: `/datasets/images/apache_spark/spark_r.sif`
+
+## Setup
+
+1. Clone the repository:
+    ```
+    $ git clone https://github.com/NREL/HPC.git
+    ```
+
+2. Change to a directory in `/scratch/$USER`.
+
+3. Add the `spark_scripts` directory to your path.
+    ```
+    $ export PATH=$PATH:<your-repo-path>/HPC/applications/spark/spark_scripts
+    ```
+
+## Compute Nodes
+Consider the type of compute nodes to acquire. If you will be performing large shuffles then
+you must get nodes with fast local storage. A minimum requirement is approximately 500 MB/s.
+
+- Any modern SSD should be sufficient.
+- A spinning disk will be too slow.
+- The Lustre filesystem on Kestrel may be fast enough for some queries.
+- The Lustre filesystem on Eagle is too slow.
+- A RAM drive (`/dev/shm`) will work well for smaller data sizes. Half the node memory is available
+on NREL compute nodes.
+
+### Kestrel
+Standard nodes do not have local storage and are unsuitable if you will shuffle data. By default,
+Spark will write to its tmp filesystem and that will quickly run out of space.
+
+Kestrel has 256 nodes with local SSDs. Pick those if you can. For example, `salloc --tmp=1600G`.
```

Comment:
I think we should start this section with this. there is no concept of "standard" nodes
```suggestion
Kestrel only has 256 nodes with local SSDs out of the 2144 CPU-only nodes available. 
Pick those if you can. For example, `salloc --tmp=1600G`.
```